### ISSUE TITLE
'npm install' and Visual C++ error

### ISSUE THREAD

**Issue by watdafox** [mafintosh/torrent-stream#137]
because of `bufferutil` and `utf-8-validate`, I can't build torrent-stream on my machine. It complains about MSB3428 (visual c++) when triggering node-gyp.

Those deps are apparently "optionnal", whatever it means, so idk if it affects the way torrent-stream is supposed to work.

Additonnal info: both deps are used by `torrent-discovery\bittorrent-tracker\ws`

**Comment by Ivshti** [comment]
These problems are related to recent changes making torrent-stream and webtorrent using unified dependencies. This is the websocket module of bittorrent-tracker. It should also be used by the peer/swarm module.

ping @feross

**Comment by feross** [comment]
`bittorrent-tracker` has depended on the `ws` module (and thus it's native dependencies) for a long time now, but torrent-stream was using an older version of bittorrent-tracker. The newer version came in when we started depending on torrent-discovery.

When optional dependencies fail to install, everything should work just fine despite npm's ridiculous error spewing. I opened an issue on bittorrent-tracker to look at removing the dependency on ws.
